test/images/network-tester/webserver.go

```diff
@@ -214,9 +215,27 @@ func main() {
 
 // Find all sibling pods in the service and post to their /write handler.
 func contactOthers(state *State) {
-	const waitTimeout = 5 * time.Minute
+	sleepTime := 5 * time.Second
+	// In large cluster getting all endpoints is pretty expensive.
+	// Thus, we will limit ourselves to send on average at most 10 such
+	// requests per second
+	if sleepTime < time.Duration(*peerCount / 10) * time.Second {
+		sleepTime = time.Duration(*peerCount / 10) * time.Second
+	}
+	timeout := 5 * time.Minute
+	// Similarly we need to bump timeout so that it is reasonable in large
+	// clusters.
+	if timeout < time.Duration(*peerCount) * time.Second {
+		timeout = time.Duration(*peerCount) * time.Second
+	}
 	defer state.doneContactingPeers()
-	client, err := client.NewInCluster()
+
+	config, err := restclient.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Unable to create config; error: %v\n", err)
+	}
+	config.ContentType = "application/vnd.kubernetes.protobuf"
```

Comment:
We already switched on protobuf by default in k8s components. We can simply switch it on here too.